From 3371be9130b4601c037e67acddaf26f8812fc6b7 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 19 Jan 2021 16:21:22 +0800
Subject: [PATCH 16/16] HACK: Use the same API version for egl decoration
 context

Somehow libmali doesn't like creating egl context for multiple API
versions.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 .../client/wayland-egl/qwaylandglcontext.cpp                  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp b/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp
index 78fe3e36..a00e21cb 100644
--- a/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp
+++ b/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp
@@ -311,8 +311,8 @@ QWaylandGLContext::QWaylandGLContext(EGLDisplay eglDisplay, QWaylandDisplay *dis
     }
 
     // Create an EGL context for the decorations blitter. By using a dedicated context we don't need to make sure to not
-    // change the context state and we also use OpenGL ES 2 API independently to what the app is using to draw.
-    QVector<EGLint> eglDecorationsContextAttrs = { EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE };
+    // change the context state.
+    QVector<EGLint> eglDecorationsContextAttrs = { EGL_CONTEXT_CLIENT_VERSION, format.majorVersion(), EGL_NONE };
     m_decorationsContext = eglCreateContext(m_eglDisplay, m_config, m_context, eglDecorationsContextAttrs.constData());
     if (m_decorationsContext == EGL_NO_CONTEXT)
         qWarning("QWaylandGLContext: Failed to create the decorations EGLContext. Decorations will not be drawn.");
-- 
2.20.1

